packages/mobile/src/pincode/utils.ts

```diff
@@ -1,30 +1,14 @@
-import { getContractKitOutsideGenerator } from 'src/web3/contracts'
-import { readPrivateKeyFromLocalDisk } from 'src/web3/privateKey'
+import { UNLOCK_DURATION } from 'src/geth/consts'
+import { getWalletAsync } from 'src/web3/contracts'
 
 export const PIN_LENGTH = 6
 
 export function isPinValid(pin: string) {
   return pin.length === PIN_LENGTH
 }
 
-export function isPinCorrect(
-  pin: string,
-  fornoMode: boolean,
-  currentAccount: string
-): Promise<typeof pin> {
-  return new Promise((resolve, reject) => {
-    if (fornoMode) {
-      readPrivateKeyFromLocalDisk(currentAccount, pin)
-        .then(() => resolve(pin))
-        .catch(reject)
-    } else {
-      getContractKitOutsideGenerator()
-        .then((contractKit: any) =>
-          contractKit.web3.eth.personal
-            .unlockAccount(currentAccount, pin, 1)
-            .then((result: boolean) => (result ? resolve(pin) : reject()))
-        )
-        .catch(reject)
-    }
-  })
+export async function isPinCorrect(pin: string, currentAccount: string): Promise<typeof pin> {
```

Comment:
Can we rename this to something like `attemptWalletUnlock` or `ensureCorrectPin`? Seems closer to what it does.
Not your code, I know, but while we're here :)